### PR TITLE
feat: Provide tooltips explaining meaning of each header

### PR DESCRIPTION
**Pull request by rhystmills**
<blockquote>"At the moment, in our new Typerighter UI, we have a single word to describe each category. There's still plenty of room for ambiguity here!"</blockquote>

Users have found the heading titles to not be as explanatory as they could be of the purpose of the checks in prosemirror-typerighter. They may wonder what the distinction between 'Amend' and 'Review' is, or why 'OK' matches are being highlighted.

This PR adds tooltips to the match headers, where we can explain what they mean in more detail.

I've used some provisional wording, but it would be good to get some consensus on what useful wording would be for each category. It would be good to have some feedback on the current wording.

This is:
| Match type | Wording |
| --- | --- |
| Amend | "Typerighter doesn’t think these matches are correct." |
| Review | "Typerighter can’t figure out whether these matches are correct or not. They might be worth reviewing." |
| OK | "Typerighter thinks these matches are correct." |

## How to test
1. Run `prosemirror-typerighter` and `typerighter` locally, according to the instructions in their READMEs. You'll also need credentials to run them locally - these can be shared by running another project with authentication locally, for example `flexible-content`.
2. Open this application, and click 'Check'. There should be a tooltip in each match type header (i.e. Amend, Review, OK) - and hovering over this should show a tooltip with relevant wording.
    - Does the styling make sense in the context of Composer, alongside its other tooltips and general design?
    - Does the wording of the descriptions make sense?

### CODE REVIEW

**Review by davidfurey — COMMENTED** [review]
It seems like logic for the tooltip is scattered between TooltipIcon, MatchHeader and SidebarMatches.  I wonder if it is possible to put most of the responsibility in one place.  Happy to discuss.

File: src/ts/components/Tooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+  const makeOpaque = () => {
+    setOpaque(true);
+  };
+
+  const [thisRef, setThisRef] = useState<HTMLElement | null>(null);
+
+  const handleMouseEnter = () => {
+    setReferenceElement(thisRef);
+    updateValues();
+    makeOpaque();
```
[the remark this answers is not in the document]
Comment (rhystmills) [reply src/ts/components/Tooltip.tsx:191]: Cheers David. I'll remove `makeOpaque` - it was hanging around from an earlier commit from a time when it did something more complex.

I'll see what I can do to keep the responsibility in one place.

[the remark this answers is not in the document]
Comment (rhystmills) [reply src/ts/components/Tooltip.tsx]: (I've moved more of the logic out of the icon component)

**Review by dskamiotis — APPROVED** [review]
Tested Locally and in CODE- working as a expected. In addition the text description is also appropriate for now - again open to further discussion.
Great css work 🎊

**Pull request merged**